### PR TITLE
CI: add non-blocking run against the linters tip versions (SC-1119)

### PR DESCRIPTION
**Pull request by paride**
## Proposed Commit Message

```
CI: add non-blocking run against the linters tip versions

Changes:
 * Tox: the tip-black and tip-isort environments.
 * Tox: reuse the non-tip commands in the tip- environments.
 * Add non-blocking (continue-on-error: true) step in the test
   test matrix to run the tip- linters on ubuntu-latest.
 * Code fixups to make tip-pylint pass.
 * Bump the pinned pylint version to 2.13.9 (last version
   supporting Python 3.6).
```

## Additional Context

Spotted by Jenkins job `cloud-init-style-tip`.

## Checklist:

 - [x] My code follows the process laid out in [the documentation](https://cloudinit.readthedocs.io/en/latest/topics/contributing.html)
 - [x] I have updated or added any unit tests accordingly
 - [ ] I have updated or added any documentation accordingly

### CODE REVIEW

**Comment by paride** [comment]
Hmm, pylint 2.13.9 is the last release supporting Python 3.6 (https://github.com/PyCQA/pylint/releases/tag/v2.13.9). We could leave it pinned to that version and stop looking at failures against tip-pylint, but this means accumulating cases that will fail against newer pylint versions, making it difficult to move ahead when time will come. I'll think of alternatives, marking this as a WIP for now.

File: tests/unittests/test_url_helper.py
Comment (paride) [reply tests/unittests/test_url_helper.py:285]: This is way better, thanks!

**Comment by paride** [comment]
I think this is ready to be reviewed. The idea is to make `tip-*` failures visible, but not blocking. I hope `continue-on-error: true` will do this, see

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepscontinue-on-error=

If we like this approach we could then remove the `cloud-init-style-tip` Jenkins jobs, as it would be redundant.

**Review by aciba90 — APPROVED** [review]
LGTM! Thanks, @paride, for improving `cloud-init` CI and bumping `pytest`'s version.

**Comment by paride** [comment]
Thanks Alberto, I added another commit to make the job names more clear, so it's easy to see if a failing job is for tip versions and should be taken as a warning and not an error.

**Comment by TheRealFalcon** [comment]
@paride is there a reason you want to do this with Github actions as opposed to Jenkins? The main reason I ask is that Github Actions feels more ephemeral. If there's a non-blocking failure on a PR, I'm probably going to forget about it as soon as I merge the PR. In Jenkins we have the big red/yellow status showing us that the job has failed until we go fix it.

File: cloudinit/config/cc_ssh.py
Comment (TheRealFalcon) [inline cloudinit/config/cc_ssh.py]:
```suggestion
                    'Skipping %s ssh_keys entry: "%s"', reason, key
```

**Review by TheRealFalcon — APPROVED** [review]
I took the liberty of pushing a change to a weird string format, but otherwise LGTM. Will merge if checks pass.

**Comment by paride** [comment]
> I took the liberty of pushing

Well done :-)

The Travis failure looks completely unrelated, and it's unclear to me I rebased on main, let's see.

**Pull request merged**
